pkg/controller/metrics/diverters/MetricsDispatcher.go

```diff
@@ -0,0 +1,55 @@
+package diverters
```

Comment:
I don't like the name "diverter".  Please change it to "adapter".

Reply:
Also please change all letters to lower for directory and change the type of file name to `metrics_dispatcher.go`

Reply:
refactored and renamed